pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -495,4 +499,104 @@ public IdealState apply(@Nullable IdealState idealState) {
       String tenant) {
     return new HashSet<>(getInstancesConfigsWithTag(instanceConfigs, TagNameUtils.getBrokerTagForTenant(tenant)));
   }
+
+  /**
+   * Add default tags to instance if the instance has no tags Note for the getDefaultTags: it is a
+   * lambda so it may not be invoked if instance already has tags. * For example () ->
+   * ImmutableList.of("Default_tenant")
+   *
+   * @param helixManager The helix manager for update
+   * @param instanceId the Helix instance Id
+   * @param getDefaultTags the default tags lambda. Something like () -> ImmutableList.of("Default_tenant") to provide default tags
+   */
+  public static void addDefaultTags(
+      HelixManager helixManager, String instanceId, Supplier<List<String>> getDefaultTags) {
+    InstanceConfig instanceConfig = getInstanceConfig(helixManager, instanceId);
+    // Add default instance tags if not exist
+    List<String> instanceTags = instanceConfig.getTags();
+    if (instanceTags == null || instanceTags.size() == 0) {
+      List<String> defaultTags = getDefaultTags == null ? null : getDefaultTags.get();
+      if (!CollectionUtils.isEmpty(defaultTags)) {
+        defaultTags.forEach(instanceConfig::addTag);
+        LOGGER.info("Updating instance tags {} for instance: {}", instanceTags, instanceId);
```

Comment:
`instanceTags` is always empty. We should log `defaultTags` instead
```suggestion
        LOGGER.info("Updating instance: {} with default tags: {}", instanceId, defaultTags);
```

Reply:
Changed too.